### ISSUE TITLE
Flow doesn't accept comments within `with`

### ISSUE THREAD

**Issue by kress95** [HackerExperience/Helix#150]
Code:

```elixir
flowing do
  with \
    # I'm a cute comment explaining a complex flow
    {:ok, _} <- {:ok, :whatever}
  do
    :ok
  end
end
```

Expected:

```elixir
:ok
```

Got:

```
** (SyntaxError) syntax error before: do
    (elixir) lib/code.ex:370: Code.require_file/2
    (elixir) lib/kernel/parallel_require.ex:57: anonymous fn/2 in Kernel.ParallelRequire.spawn_requires/5
```

The code works as expected without the comment.

**Comment by mememori** [comment]
This is not actually an issue with Flow but with elixir.

Let me explain why:

With generates a series of nested case clauses based on the input values. So basically, your with becomes `with({:ok, _} <- ..., [do: ...])`.

Elixir's tokenizer has to assume by the context when a function call (or binding call) ends.

The code you wrote means to the tokenizer vaguely the following:

```elixir
with()
# # I'm a cute comment explaining a complex flow
{:ok, _} <- {:ok, :whatever}
do :ok end
```

As you can see, it has no real meaning because, as the `with` didn't receive the first param, (nor any tip that the function call continued on next line) it had do assume that you were calling `with/0` and then later executing the rest as a separate thing. Shall you have run the code excerpt you provided directly in terminal, you would see the following:

```
iex(1)> flowing do
...(1)>   with \
...(1)>     # I'm a cute comment explaining a complex flow
...(1)>     {:ok, _} <- {:ok, :whatever}
...(1)>   do
...(1)>     :ok
...(1)>   end
...(1)> end
** (SyntaxError) iex:5: syntax error before: do
    
iex(1)>   with \
...(1)>     # I'm a cute comment explaining a complex flow
warning: variable "with" does not exist and is being expanded to "with()", please use parentheses to remove the ambiguity or change the variable name
  iex:1

** (CompileError) iex:1: undefined function with/0
    
iex(1)>     {:ok, _} <- {:ok, :whatever}
** (CompileError) iex:1: undefined function <-/2
    
iex(1)>   do
...(1)>     :ok
...(1)>   end
** (SyntaxError) iex:1: syntax error before: do
```

As an example to prove that this is not related to Flow:
```
iex(1)> case \
...(1)>   # comment
warning: variable "case" does not exist and is being expanded to "case()", please use parentheses to remove the ambiguity or change the variable name
  iex:1

** (CompileError) iex:1: undefined function case/0
    
iex(1)>   true
true
iex(2)> do
...(2)>   _ -> :ok
...(2)> end
** (SyntaxError) iex:2: syntax error before: do
```

**Comment by mememori** [comment]
Also, the reason you're only seeing the `do` warning is because Flow is a macro that is manipulating the tokenized AST. Basically it received an invalid AST (as i shown before, it understands that the `do` block does not belongs to any function), so for this very reason you don't get an error saying that `with/0` doesn't exists or that those pattern clauses are wrong (for the simple reason that they were not checked yet, the first thing the compiler sees in this case is a out-of-place do block)

So, tl;dr, is invalid and wontfix; just avoid using comments on ambiguous places
